src/shell/directory_stack.rs

```diff
@@ -155,11 +204,76 @@ impl DirectoryStack {
         self.dirs.truncate(DirectoryStack::get_size(variables));
     }
 
-    pub fn dirs<I: IntoIterator>(&self, _: I) -> i32
+    fn insert_dir(&mut self, index: usize, path: PathBuf, variables: &Variables) {
+        self.dirs.insert(index, path);
+        self.dirs.truncate(DirectoryStack::get_size(variables));
+    }
+
+    pub fn dirs<I: IntoIterator>(&mut self, args: I) -> i32
         where I::Item: AsRef<str>
     {
-        self.print_dirs();
-        SUCCESS
+	const CLEAR: u8 = 1;		// -c
+	const ABS_PATHNAMES: u8 = 2;	// -l
+	const MULTILINE: u8 = 4;	// -p | -v
+	const INDEX: u8 = 8;		// -v
```

Comment:
I'd recommend using [`bitflags`](https://docs.rs/bitflags/0.9.1/bitflags/) over doing this manually.